lib/utils.pm

```diff
@@ -312,15 +312,24 @@ sub minimal_patch_system {
     }
 }
 
+=head2 workaround_type_encrypted_passphrase
+
+    workaround_type_encrypted_passphrase()
+
+Record soft-failure for unresolved feature which we think is important and then unlock encrypted boot partitions if we expect it to be encrypted. This condition is met on 'storage-ng' which by default puts the boot partition within the encrypted LVM same as in test scenarios where we explicitly create an LVM including boot (C<FULL_LVM_ENCRYPT>). C<ppc64le> was already doing the same by default also in the case of pre-storage-ng but not anymore for storage-ng.
+
+=cut
 sub workaround_type_encrypted_passphrase {
-    if (
-        get_var('FULL_LVM_ENCRYPT')
-        || (check_var('ARCH', 'ppc64le')
-            && (get_var('ENCRYPT') && !get_var('ENCRYPT_ACTIVATE_EXISTING') || get_var('ENCRYPT_FORCE_RECOMPUTE'))))
-    {
-        record_soft_failure 'workaround https://fate.suse.com/320901' if sle_version_at_least('12-SP4');
-        unlock_if_encrypted;
-    }
+    # nothing to do if the boot partition is not encrypted in FULL_LVM_ENCRYPT
+    return if get_var('UNENCRYPTED_BOOT');
+    return if !get_var('ENCRYPT') && !get_var('FULL_LVM_ENCRYPT');
+    # ppc64le is always doing the opposite by default :)
+    return if (!is_storage_ng && !get_var('FULL_LVM_ENCRYPT')) || (is_storage_ng && check_var('ARCH', 'ppc64le'));
```

Comment:
ppc64le pre-storage-ng missing now

```perl
return if !is_storage_ng && !check_var('ppc64le') && !get_var('LVM_FULL_ENCRYPT');
return if is_storage_ng && check_var('ppc64le') && !get_var('LVM_FULL_ENCRYPT');
```

Reply:
```perl
if (!get_var('LVM_FULL_ENCRYPT')) {
  return if (!is_storage_ng || check_var('ARCH', 'ppc64le'));
}
```

^- created by 3 dead brains

Reply:
https://math.stackexchange.com/questions/1272958/disjunctive-normal-form-and-shannon-normal-form I will just leave it here

Reply:
I suggest following if want to change, otherwise having DNF is good enough as easy to read and understand.

```
if(is_storage_ng) {
  return if check_var('ARCH', 'ppc64le');
} else {
  return unless get_var('FULL_LVM_ENCRYPT');
}
```

Reply:
done